### DISCUSSION TITLE
Code-server shows blank page on iPad, works on MacBook (local network)

### DISCUSSION THREAD

**Discussion by mohdyusuf** [coder/code-server#2512] [Q&A] [answered]
I followed the installation instructions from here:
https://github.com/cdr/code-server/blob/v3.8.0/doc/guide.md#lets-encrypt

I was able to see the login page on iPad, and once I put in password and logged in, the page is blank and not loading.
Followed the instructions as stated, except on the domain, I used "pi.local" which I can access on my local network.

I tried accessing https://pi.local from my Mac and it works fine, I can login and see the VSCODE.

Is there something that I missed?

I also read this, but can someone clarify on "exposing" code-server to `0.0.0.0`?
https://github.com/cdr/code-server/discussions/1789

Edit:
Btw, I am using Caddy server route. Any suggestion on how to debug this?

**Comment by code-asher** [comment]
Exposing on 0.0.0.0 means to do something like:

```
code-server --host 0.0.0.0
```

This makes code-server listen to requests that come from outside the
machine.

But since you're using Caddy that won't be necessary since Caddy handles
the external requests.

If you are trying to access `https://pi.local` that might be why it's
not working. iPads are rather strict about certificates (at least when
it comes to websockets, which is why you can see the login page but not
VS Code) and you need to have a real domain with a certificate signed by
a certificate authority.

But we have some docs on getting self-signed certificates working if
needed: https://github.com/cdr/code-server/blob/v3.8.0/doc/guide.md#self-signed-certificate

**Reply by mohdyusuf** [reply]
Thanks @code-asher  for the reply. I am new to all of these, and your reply is opening up some of my questions w/r/t Caddy and listening to port etc.

I did read thru the self-signed section - for some reasons it mentioned that it wont work with iPad in general and I didnt try.

"Self Signed Certificate
note: Self signed certificates do not work with iPad normally. See ./ipad.md for details.

We recommend this as a last resort because self signed certificates do not work with iPads and can cause other bizarre issues. Not to mention all the warnings when you access code-server."

After re-reading this, perhaps I could try again for signing. I am trying to just have this working on local network for now, will try to see if I can get that to work.

**Reply by code-asher** [reply]
Ahhh yeah the wording makes it sound like it's impossible. I do believe it's possible but it requires a bunch of steps. I haven't tested personally though.

**Reply by mohdyusuf** [reply]
Great, thanks! I will give it a try.

**Comment by mohdyusuf** [comment, marked as answer]
Now that @code-asher has given hints, I tried a little more and finally was able to get `code-server` on iPad working.

Here’s the steps:
* follow the installation guideline and use the “Self Signed Certificate” route. I didnt need Caddy/NGINX
* Follow this iPad specific guideline:
  * https://github.com/cdr/code-server/blob/v3.8.0/doc/ipad.md#how-to-access-code-server-with-a-self-signed-certificate-on-ipad
  * Note, for me use this to find the log and `.crt` path of `code-server` on the server `sudo systemctl status code-server@$USER`
 
Hope this helps someone in the future.
